File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int i = 0;
        int j = 0;
        int idx = 0;
        int[] tempArr = new int[m+n];
        while(i<m && j<n) {
            if(nums1[i] < nums2[j]) {
                tempArr[idx] = nums1[i];
                i++;
                idx++;
            } else {
                tempArr[idx] = nums2[j];
                j++;
                idx++;
            }
        }
        while(i<m) {
            tempArr[idx] = nums1[i];
            i++;
            idx++;
        }

        while(j<n) {
            tempArr[idx] = nums2[j];
            j++;
            idx++;
        }

        for(int x=0; x<tempArr.length; x++) {
            nums1[x] = tempArr[x];
        }
    }
}

File: 88-merge-sorted-array/merge-sorted-array.cs
public class Solution {
    public void Merge(int[] nums1, int m, int[] nums2, int n) {
        if(n==0) return;
        int nums1Len = nums1.Length-1;
        int nums1Idx = nums1Len - n;
        int nums2Idx = n-1;
        while(nums2Idx >= 0 && nums1Idx >= 0) {
            if(nums1[nums1Idx] <= nums2[nums2Idx]) {
                nums1[nums1Len] = nums2[nums2Idx];
                nums1Len--;
                nums2Idx--;
            } else {
                nums1[nums1Len] = nums1[nums1Idx];
                nums1[nums1Idx] = 0;
                nums1Idx--;
                nums1Len--;
            }
        }

        while(nums2Idx >= 0) {
            nums1[nums1Len] = nums2[nums2Idx];
            nums2Idx--;
            nums1Len--;
        }
    }
}